ethereum/statetransition/src/main/java/tech/pegasys/artemis/statetransition/AttestationAggregator.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.statetransition;
+
+import static tech.pegasys.artemis.datastructures.util.AttestationUtil.getAttesterIndexIntoCommittee;
+import static tech.pegasys.artemis.datastructures.util.AttestationUtil.representsNewAttester;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.primitives.UnsignedLong;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.tuweni.bytes.Bytes32;
+import tech.pegasys.artemis.datastructures.operations.AggregateAndProof;
+import tech.pegasys.artemis.datastructures.operations.Attestation;
+import tech.pegasys.artemis.datastructures.validator.AggregatorInformation;
+import tech.pegasys.artemis.datastructures.validator.AttesterInformation;
+import tech.pegasys.artemis.util.bls.BLSAggregate;
+import tech.pegasys.artemis.util.bls.BLSSignature;
+
+public class AttestationAggregator {
+
+  private final ConcurrentHashMap<Bytes32, Attestation> dataHashToAggregate =
+      new ConcurrentHashMap<>();
+
+  @VisibleForTesting
+  final Map<UnsignedLong, AggregatorInformation> committeeIndexToAggregatorInformation =
+      new ConcurrentHashMap<>();
+
+  private final Map<UnsignedLong, List<Attestation>> committeeIndexToAggregatesList =
```

Comment:
This doesn't need to be a list does it?
```suggestion
  private final Map<UnsignedLong, Attestation> committeeIndexToAggregatesList =
```

Reply:
good catch. done.